cbor/cbor.go

```diff
@@ -752,9 +752,11 @@ func (d *Decoder) decodeArray(rv reflect.Value, additional []byte) error {
 		return d.decodeArrayToStruct(rv, additional)
 	case reflect.Slice, reflect.Array:
 		return d.decodeArrayToSlice(rv, additional)
+	case reflect.Map:
+		return d.decodeMap(rv, additional)
```

Comment:
This is new functionality and needs explicit tests.

Is decoding an array into a map something we really want, though? It seems to violate the principle of least surprise. If you try to decode an array and provide a map pointer, is that more likely intended or a mistake?

It's also just not that hard to convert a slice of pairs into a map with Go code after decoding... If nothing else, we may want this to be a configurable option of the decoder that is off by default.

Where is this functionality required in this PR?